### ISSUE TITLE
`accept-comment-above-attributes = true` doesn't skip `cfg_attr` attributes

### ISSUE THREAD

**Issue by joshlf** [rust-lang/rust-clippy#16087]
### Summary

When `accept-comment-above-attributes = true` is set, Clippy properly ignores most attributes, but does not ignore `cfg_attr` attributes.

### Lint Name

undocumented_unsafe_blocks

### Reproducer

I tried this code (with `accept-comment-above-attributes = true`):

```rust
// SAFETY:
// - `Unalign<T>` promises to have alignment 1, and so we don't require that `T:
//   Unaligned`.
// - `Unalign<T>` has the same bit validity as `T`, and so it is `FromZeros`,
//   `FromBytes`, or `IntoBytes` exactly when `T` is as well.
// - `Immutable`: `Unalign<T>` has the same fields as `T`, so it contains
//   `UnsafeCell`s exactly when `T` does.
// - `TryFromBytes`: `Unalign<T>` has the same the same bit validity as `T`, so
//   `T::is_bit_valid` is a sound implementation of `is_bit_valid`.
//
#[cfg_attr(
    all(__ZEROCOPY_INTERNAL_USE_ONLY_NIGHTLY_FEATURES_IN_TESTS, any(feature = "derive", test)),
    expect(unused_unsafe)
)]
#[cfg_attr(
    all(
        not(__ZEROCOPY_INTERNAL_USE_ONLY_NIGHTLY_FEATURES_IN_TESTS),
        any(feature = "derive", test)
    ),
    allow(unused_unsafe)
)]
const _: () = unsafe {
```

I saw this happen:

```
error: unsafe block missing a safety comment
   --> src/wrappers.rs:143:15
    |
143 | const _: () = unsafe {
    |               ^^^^^^^^
    |
    = help: consider adding a safety comment on the preceding line
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#undocumented_unsafe_blocks
note: the lint level is defined here
   --> src/lib.rs:272:5
    |
272 |     clippy::undocumented_unsafe_blocks,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

I expected to see this happen: Nothing

### Version

```text
rustc 1.93.0-nightly (20383c9f1 2025-11-03)
binary: rustc
commit-hash: 20383c9f1d84eb9b9c6668a1668ef68a81eae274
commit-date: 2025-11-03
host: x86_64-unknown-linux-gnu
release: 1.93.0-nightly
LLVM version: 21.1.3
```

**Comment by zihan0822** [comment]
probably duplicate of #15915
